flake.nix

```diff
@@ -0,0 +1,156 @@
+{
+  description = "Shadysim";
+
+  inputs = {
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.05";
+
+    pycrypto-src = {
+      url = "github:pycrypto/pycrypto";
+      flake = false;
+    };
+
+    simtools-src = {
+      url = "git+https://git.osmocom.org/sim/sim-tools";
+      flake = false;
+    };
+
+    imsi-pseudo-src = {
+      url = "git+https://git.osmocom.org/imsi-pseudo";
+      flake = false;
+    };
+
+  };
+
+
+  outputs =
+    { self
+    , nixpkgs
+    , pycrypto-src
+    , simtools-src
+    , imsi-pseudo-src
+    , ...
+    }:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }
+      );
+
+    in
+    {
+
+      overlay = final: prev: rec {
+
+        pycrypto =
+          with final.python2Packages;
+          buildPythonPackage {
+            pname = "pycrypto";
+            version = "2.6.1";
+            src = pycrypto-src;
+          };
+
+        shadysim =
+          with final.python2Packages;
+          buildPythonApplication {
+            pname = "shadysim";
+            version = "0.1.0";
+            src = "${simtools-src}/shadysim";
+
+            postPatch = ''
+              cat << EOF > setup.py
+              #!/usr/bin/env python
+
+              from setuptools import setup, find_packages
+
+              setup(
+                  name="shadysim",
+                  version="1.0",
+                  # Modules to import from other scripts:
+                  packages=find_packages(),
+                  # Executables
+                  scripts=["shadysim.py"],
+              )
+              EOF
+            '';
+
+            postInstall = ''
+              mv $out/bin/shadysim.py $out/bin/shadysim
```

Comment:
my heart is mind with clarity :-)
The hackers probably do it the other way, feel free to keep using it.